### PR TITLE
REF: Make CategoricalIndex comparison defer to Categorical comparison

### PR DESCRIPTION
**Pull request by jbrockmendel**
Partially addresses #19513.

After this, CategoricalIndex will be defining comparison ops identically to DTA/TDA/PA, could share some code.

### CODE REVIEW

**Comment by jreback** [comment]
looks fine; are the added paths  hit in coverage? e.g. do we need any additional tests?

**Comment by jbrockmendel** [comment]
The new paths are all hit, yes.

**Comment by jreback** [comment]
thanks @jbrockmendel

File: pandas/core/ops/__init__.py [hunk 1]
```diff
@@ -1038,8 +1038,14 @@ def wrapper(self, other, axis=None):
             # Defer to DataFrame implementation; fail early
             return NotImplemented
 
-        elif isinstance(other, ABCSeries) and not self._indexed_same(other):
+        if isinstance(other, ABCSeries) and not self._indexed_same(other):
             raise ValueError("Can only compare identically-labeled Series objects")
+        elif (
+            is_list_like(other)
+            and len(other) != len(self)
+            and not isinstance(other, (set, frozenset))
```
Comment (jorisvandenbossche) [inline pandas/core/ops/__init__.py]: @jbrockmendel maybe by accident due to several PRs that were related / rebase after each other, but this seemingly introduced again what you removed in https://github.com/pandas-dev/pandas/pull/27873 (although the other PR was merged later ..)

No matter how it happened ;), the check is still in master ..

Comment (jbrockmendel) [reply pandas/core/ops/__init__.py]: yah this looks like a rebase screwup, will address

**Pull request merged**